Sources/SourceKitLSP/Swift/SwiftLanguageServer.swift

```diff
@@ -161,6 +161,71 @@ public final class SwiftLanguageServer: ToolchainLanguageServer {
     }
   }
 
+  /// Update lexical and syntactic tokens for the given `snapshot`.
+  func updateLexicalAndSyntacticTokens(
+    response: SKDResponseDictionary,
+    for snapshot: DocumentSnapshot
+  ) {
+    let uri = snapshot.document.uri
+
+    if let syntaxMap: SKDResponseArray = response[keys.syntaxmap] {
```

Comment:
Just to clarify, the syntax map is providing delta updates whereas `substructure` is always a full update, right?

Reply:
Yes, that is correct, as far as I understood it.